test/testmachinery/shoots/applications/shoot_app.go

```diff
@@ -86,8 +87,11 @@ var _ = ginkgo.Describe("Shoot application testing", func() {
 		}
 
 		url := fmt.Sprintf("https://api.%s/api/v1/namespaces/%s/services/https:kubernetes-dashboard:/proxy", *f.Shoot.Spec.DNS.Domain, "kubernetes-dashboard")
-		dashboardToken, err := framework.GetObjectFromSecret(ctx, f.GardenClient, f.Shoot.Namespace, f.Shoot.Name+"."+gutil.ShootProjectSecretSuffixKubeconfig, "token")
+		kubeConfig, err := framework.GetObjectFromSecret(ctx, f.GardenClient, f.Shoot.Namespace, f.Shoot.Name+"."+gutil.ShootProjectSecretSuffixKubeconfig, framework.KubeconfigSecretKeyName)
 		framework.ExpectNoError(err)
+		client, err := kubernetes.NewClientFromBytes([]byte(kubeConfig))
+		framework.ExpectNoError(err)
+		dashboardToken := client.RESTConfig().BearerToken
```

Comment:
We don't need to create a new client just to fetch the token from the kubeconfig.

```suggestion
		kubeconfigData, err := framework.GetObjectFromSecret(ctx, f.GardenClient, f.Shoot.Namespace, f.Shoot.Name+"."+gutil.ShootProjectSecretSuffixKubeconfig, framework.KubeconfigSecretKeyName)
		framework.ExpectNoError(err)

		kubeconfig := &clientcmdv1.Config{}
		_, _, err = clientcmdlatest.Codec.Decode([]byte(kubeconfigData), nil, kubeconfig)
		framework.ExpectNoError(err)

		Expect(kubeconfig.AuthInfos).To(HaveLen(1))
		Expect(kubeconfig.AuthInfos[0].AuthInfo.Token).NotTo(BeEmpty())
		dashboardToken := kubeconfig.AuthInfos[0].AuthInfo.Token
```